python/lsst/ip/diffim/getTemplate.py

```diff
@@ -474,3 +479,219 @@ def runDataRef(self, *args, **kwargs):
 
     def runQuantum(self, **kwargs):
         raise NotImplementedError("Calexp template is not supported with gen3 middleware")
+
+
+class GetMultiTractCoaddTemplateConnections(pipeBase.PipelineTaskConnections,
+                                            dimensions=("instrument", "visit", "detector", "skymap"),
+                                            defaultTemplates={"coaddName": "goodSeeing",
+                                                              "warpTypeSuffix": "",
+                                                              "fakesType": ""}):
+    bbox = pipeBase.connectionTypes.Input(
+        doc="BBoxes of calexp used determine geometry of output template",
+        name="{fakesType}calexp.bbox",
+        storageClass="Box2I",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    wcs = pipeBase.connectionTypes.Input(
+        doc="WCSs of calexps that we want to fetch the template for",
+        name="{fakesType}calexp.wcs",
+        storageClass="Wcs",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    skyMap = pipeBase.connectionTypes.Input(
+        doc="Input definition of geometry/bbox and projection/wcs for template exposures",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        dimensions=("skymap", ),
+        storageClass="SkyMap",
+    )
+    # TODO DM-31292: Add option to use global external wcs from jointcal
+    # Needed for DRP HSC
+    coaddExposures = pipeBase.connectionTypes.Input(
+        doc="Input template to match and subtract from the exposure",
+        dimensions=("tract", "patch", "skymap", "band"),
+        storageClass="ExposureF",
+        name="{fakesType}{coaddName}Coadd{warpTypeSuffix}",
+        multiple=True,
+        deferLoad=True
+    )
+    outputExposure = pipeBase.connectionTypes.Output(
+        doc="Warped template used to create `subtractedExposure`.",
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="ExposureF",
+        name="{fakesType}{coaddName}Diff_templateExp{warpTypeSuffix}",
+    )
+
+
+class GetMultiTractCoaddTemplateConfig(pipeBase.PipelineTaskConfig, GetCoaddAsTemplateConfig,
+                                       pipelineConnections=GetMultiTractCoaddTemplateConnections):
+    warp = pexConfig.ConfigField(
+        dtype=afwMath.Warper.ConfigClass,
+        doc="warper configuration",
+    )
+    coaddPsf = pexConfig.ConfigField(
+        doc="Configuration for CoaddPsf",
+        dtype=CoaddPsfConfig,
+    )
+
+    def setDefaults(self):
+        self.warp.warpingKernelName = 'lanczos5'
+        self.coaddPsf.warpingKernelName = 'lanczos5'
```

Comment:
DRP uses lanczos5 for CoaddPsfs during deep coaddition because DM-17429.  Option is either to put this config here slowing down AP slightly or put it in the obs_package DPR.yaml.  Either is fine with me.  Both I and Tanaka-san found that warping with 5 only takes 5% longer: DM-18295

Reply:
I think it's fine to put this here. If needed in AP, we can downgrade it in the APpipe config.